src/armadillo.mk

```diff
@@ -36,9 +36,21 @@ define $(PKG)_BUILD
      echo 'Libs: -larmadillo'; \
     ) > '$(PREFIX)/$(TARGET)/lib/pkgconfig/$(PKG).pc'
 
+    # fix cmake file, avoid absolute paths to libraries
+    $(SED) -i -e 's-\(/[^/;]\+\)\+/lib/lib\([[:alnum:]]\+\).a-\2-g' \
+                 '$(PREFIX)/$(TARGET)/share/Armadillo/CMake/ArmadilloLibraryDepends.cmake'
+
+    # fix cmake file, avoid unnecessary absolute path
+    $(SED) -i -e 's!\(ARMADILLO_LIBRARY_DIRS[ \t]\+\)[^ \t()]\+\(.*\)!\1""\2!g' \
+                 '$(PREFIX)/$(TARGET)/share/Armadillo/CMake/ArmadilloConfig.cmake'
+
+    # fix cmake file, avoid unnecessary absolute path
+    $(SED) -i -e 's!\(ARMADILLO_INCLUDE_DIRS[ \t]\+\)[^ \t()]\+\(.*\)!\1""\2!g' \
+                 '$(PREFIX)/$(TARGET)/share/Armadillo/CMake/ArmadilloConfig.cmake'
+
```

Comment:
Wouldn't it be better to do this with a patch than this sed hack?

Reply:
I've removed this part from the patch. I agree it would be nice to fix that properly upstream, so that the cmake files are generated correctly - I've opened an issue upstream. I don't know enough cmake (nor Armadillo) to be comfortable writing a patch for that.